docs: Rename Admin Roles to get rid of Partner-naming (#1075)
meshcloud/meshcloud-docs: docs/administration.emergency-users.md

title: "Guide: Emergency Users"
---

> meshStack manages access to cloud platforms, projects and resources. In case urgent intervention is required by someone without regular access permissions there must be a defined process to securely access meshProjects and associated meshTenants. This page outlines step-by-step procedures that Platform Engineers can use as the basis for their own emergency procedures. Depending on your organization's requirements, these procedures can be augmented with additional organisational or technical procedures.

Reviewer: ```suggestion
meshStack manages access to cloud platforms, projects and resources. In case urgent intervention is required by someone without regular access permissions there must be a defined process to securely access meshProjects and associated Tenants. This page outlines step-by-step procedures that users with the role "Platform Engineers" can use as the basis for their own emergency procedures. Depending on your organization's requirements, these procedures can be augmented with additional organisational or technical procedures.
```

- At the bottom of the screen, type in the name or e-mail of the new user and invite them with the desired role.

You can also grant workspace manager rights to the newly invited user, i.e. if the emergency user needs to modify other user permissions. In this case, the new user can perform the following steps themselves.

### Adding emergency as Workspace Manager

The user can then be assigned to projects belonging to the workspace:

- In the project overview in the [workspace control plane](./meshcloud.workspace.md#managing-your-meshworkspace), open the designated project by clicking on its name.
- Navigate to the **Project Access** tab and open the **Current Access** subtab.
- Add the user with the desired project role.

Since emergency access should only be temporary, it's strongly advised to assign the user a role with a [set expiration date](./meshcloud.project.md#expiry-of-a-user-assignment) which will ensure that the user is automatically removed from the project after the specified date.

### Approving emergency user requests

> User project role assignments can be configured to require consent from multiple workspace managers ([4 eye principle](./meshstack.authorization.md#user-project-role-approval)). To avoid situations where not enough workspace managers are available to confirm an urgent user role request, the Admin can confirm project role requests directly:

Reviewer 2: i: admin is written with upper case but in line 41 it's lower case again

Author: fixed

title: "Guide: Emergency Users"
---

> meshStack manages access to cloud platforms, projects and resources. In case urgent intervention is required by someone without regular access permissions there must be a defined process to securely access projects and associated tenants. This page outlines step-by-step procedures that users with the role "platform engineer" can use as the basis for their own emergency procedures. Depending on your organization's requirements, these procedures can be augmented with additional organisational or technical procedures.

Reviewer 2: ```suggestion
meshStack manages access to cloud platforms, projects and resources. In case urgent intervention is required by someone without regular access permissions there must be a defined process to securely access projects and associated tenants. This page outlines step-by-step procedures that users with the role "Platform Engineer" can use as the basis for their own emergency procedures. Depending on your organization's requirements, these procedures can be augmented with additional organisational or technical procedures.
```

### Approving emergency user requests

> User project role assignments can be configured to require consent from multiple workspace managers ([4 eye principle](./meshstack.authorization.md#user-project-role-approval)). To avoid situations where not enough workspace managers are available to confirm an urgent user role request, the meshAdmin can confirm project role requests directly:

Reviewer: ```suggestion
User project role assignments can be configured to require consent from multiple workspace managers ([4 eye principle](./meshstack.authorization.md#user-project-role-approval)). To avoid situations where not enough workspace managers are available to confirm an urgent user role request, an admin user can confirm project role requests directly:
```

> - Ensure that the admin is selected from the meshWorkspace drop down

Reviewer: ```suggestion
- Ensure that the admin user is selected from the Workspace drop down
```

- Remove user from meshProject by opening the project again and navigating to **Project Access** > **Current Access** (performed automatically if expiration date was set)
- Remove user from meshWorkspace via the [workspace control plane](./meshcloud.workspace.md#managing-your-meshworkspace): go to **Workspace Access** > **Current Access**.

> ## Access with meshAdmin

Author: These where like this in the docs am I expected to rename the mesh prefix?

> ## Access with meshAdmin

Reviewer: Yes whenever you change something in the docs please also make sure the text consideres current standards.

> ## Access with meshAdmin

Author: done

> Even when no workspace manager is available, users with organization admin/user access can manage permissions for their managed meshWorkspaces.

Reviewer: ```suggestion
Even when no workspace manager is available, admin users can manage permissions for workspaces.
```

> ### Adding emergency user as Organization Admin

Reviewer: Lets remove this whole chapter about emergency users same for the remove emergency user

Note: this only works when Workspace Manager role assignment is enabled via the panel. This depends on the `restrictCustomerAdminRoleAssignment` configuration
value ([read more here](meshstack.onboarding.md#workspace-user-invitations))

In order to manage users for a specific meshProject, the admin user must add their own account to the respective meshWorkspace as a workspace manager:

- Ensure that the meshAdmin is selected from the meshWorkspace dropdown
- Open "Administration" from the settings menu in the top right
- Navigate to "Workspaces" and select "Workspace User" from the actions column for the target workspace
- Activate “Add Myself” button in the top right

Afterwards they may proceed to manage users for this meshWorkspace as a workspace manager (see previous section).

### Removing emergency user as Organization Admin

> Since the meshAdmin user is now a workspace manager the procedure is the same as outlined in the previous section.

Reviewer 2: @chewbeccy i think this part is quiet confusing.
Shall we just use the term admin here

Author: sure. Done.

However, to revert the temporary workspace manager role assignment, another workspace manager must remove the admin user from the workspace via **Workspace Access** > **Current Access**.

## Auditing Emergency Access

> Since meshWorkspace/meshProject access permissions should not be granted lightly all changes to them are logged and can be audited by organization admin/user.

Reviewer 2: ```suggestion
Since workspace/project access permissions should not be granted lightly all changes to them are logged and can be audited by organization admin/user.
```

> ### Workspace History

Reviewer: This is no longer available please remove

- Ensure that the meshAdmin is selected from the meshWorkspace drop down
- Open "Administration" from the settings menu in the top right
- Navigate to "Workspaces" and select "Workspace History"

The list contains all meshWorkspace events (i.e. sent invitations, added/removed users, role changes), when they occurred and who initiated the action. Event specific information (i.e. who the recipient of an invite was) is available via the “Details” button.

> ### Project History

Reviewer: This is no longer available please remove
